webapp/sources/rudder/rudder-web/src/main/elm/sources/Rules/ViewUtils.elm

```diff
@@ -141,52 +160,60 @@ valueCompliance =
       let
-        sortFunction =  subItemOrder valueCompliance model sortId
+        sortFunction = subItemOrder (valueCompliance complianceFilters) model sortId
       in
-        List.sortWith sortFunction item.values
+        item.values
+        |> List.filter (filterValueByCompliance complianceFilters)
+        |> List.sortWith sortFunction
     )
     (\_ i -> i)
     [ ("Node", .nodeId >> (\nId -> span[][text (getNodeHostname mod nId.value), goToBtn (getNodeLink mod.contextPath nId.value)]),  (\d1 d2 -> N.compare d1.name d2.name))
     , ("Compliance", .complianceDetails >> buildComplianceBar ,  (\d1 d2 -> Basics.compare d1.compliance d2.compliance))
     ]
     (.nodeId >> .value)
-    (Just (\item -> showComplianceDetails valueCompliance item))
-    (always (List.map Tuple3.first valueCompliance.rows))
+    (Just (\item -> showComplianceDetails (valueCompliance complianceFilters) item))
+    (always (List.map Tuple3.first (valueCompliance complianceFilters).rows))
 
-byComponentCompliance : ItemFun  value subValue valueData -> ItemFun (ComponentCompliance value) (Either (ComponentCompliance value) value) (ComponentCompliance value)
-byComponentCompliance subFun =
+byComponentCompliance : ItemFun value subValue valueData -> ComplianceFilters -> ItemFun (ComponentCompliance value) (Either (ComponentCompliance value) value) (ComponentCompliance value)
+byComponentCompliance subFun complianceFilters =
   let
     name = \item ->
-             case item of
-               Block b -> b.component
-               Value c -> c.component
+      case item of
+        Block b -> b.component
+        Value c -> c.component
     compliance = \item ->
-                   case item of
-                     Block b -> b.complianceDetails
-                     Value c -> c.complianceDetails
+      case item of
+        Block b -> b.complianceDetails
+        Value c -> c.complianceDetails
     complianceValue = \item ->
-                   case item of
-                     Block b -> b.compliance
-                     Value c -> c.compliance
+      case item of
+        Block b -> b.compliance
+        Value c -> c.compliance
   in
-  ItemFun
+    ItemFun
     ( \item model sortId ->
       case item of
         Block b ->
           let
-            sortFunction =  subItemOrder (byComponentCompliance subFun) model sortId
+            sortFunction = subItemOrder (byComponentCompliance subFun complianceFilters) model sortId
           in
-             List.map Left (List.sortWith sortFunction b.components)
+            b.components
+            |> List.fvalueComplianceilter (filterByCompliance complianceFilters)
+            |> List.sortWith sortFunction
+            |> List.map Left
         Value c ->
           let
             sortFunction =  subItemOrder subFun model sortId
           in
-             List.map Right (List.sortWith sortFunction c.values)
+            c.values
+            -- @TODO : |> List.filter someFunction
```

Comment:
The last thing missing I think, is the filter at this level. I haven't managed to write anything functional...